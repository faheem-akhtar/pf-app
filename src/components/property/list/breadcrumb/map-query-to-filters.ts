import { FiltersParametersEnum } from 'enums/filters/parameters.enum';
import { FiltersQueryParametersEnum } from 'enums/filters/query-parameters.enum';
import { LocationService } from 'services/location/service';

const entries: string[] = Object.values(FiltersQueryParametersEnum);

// eslint-disable-next-line @typescript-eslint/no-explicit-any
type MapQueryToFiltersResultType = { [key in FiltersParametersEnum]?: any };

/**
 * @param link Clicked breadcrumb link
 */
export const breadcrumbMapQueryToFilters = (link: string): MapQueryToFiltersResultType => {
  const query = link.split('search?')[1];
  const values: MapQueryToFiltersResultType = {};

  query.split('&').forEach((param) => {
    const [queryKey, queryValue] = param.split('=');
    const indexOfParam = entries.indexOf(queryKey);
    const filtersQueryParameterKey = Object.keys(FiltersQueryParametersEnum)[indexOfParam];

    if (queryKey === FiltersQueryParametersEnum.locationsIds) {
      values[FiltersParametersEnum.locationsIds] = queryValue
        .split('-')
        ?.map((id) => LocationService.find((location) => location.id === id));
      return;
    }

    values[FiltersParametersEnum[filtersQueryParameterKey as keyof typeof FiltersParametersEnum]] = queryValue;
  });

  return values;
};
