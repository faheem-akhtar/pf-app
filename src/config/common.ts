import { ConfigCommonInterface } from 'types/config/common.interface';
import { CountryAreaUnitEnum } from 'enums/country/area-unit.enum';
import { CountryCurrencyEnum } from 'enums/country/currency.enum';
import { LanguageCodeEnum } from 'enums/language/code.enum';

export const configCommon: ConfigCommonInterface = {
  areaUnit: CountryAreaUnitEnum.sqft,
  currencyCode: CountryCurrencyEnum.aed,
  countryCode: 'ae',
  language: {
    current: LanguageCodeEnum.en,
    alternative: LanguageCodeEnum.ar,
  },
};
