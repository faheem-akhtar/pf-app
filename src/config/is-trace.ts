// The code inside configIsTrace condition will be eliminated from the bundle,
// But at the same time allowes developers to leave traces for easier
// debugging of the application
// Start app with TRACE=1 yarn dev to use it
export const configIsTrace = !!process.env.TRACE;
