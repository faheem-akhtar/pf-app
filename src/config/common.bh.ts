import { CountryAreaUnitEnum } from 'enums/country/area-unit.enum';
import { CountryCurrencyEnum } from 'enums/country/currency.enum';
import { LanguageCodeEnum } from 'enums/language/code.enum';

import { ConfigCommonInterface } from './common.interface';

export const configCommon: ConfigCommonInterface = {
  areaUnit: CountryAreaUnitEnum.sqm,
  currencyCode: CountryCurrencyEnum.bhd,
  countryCode: 'bh',
  insertDefaultLanguage: true,
  language: {
    current: LanguageCodeEnum.en,
    alternative: LanguageCodeEnum.ar,
  },
};
