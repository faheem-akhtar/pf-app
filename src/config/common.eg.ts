import { ConfigCommonInterface } from 'types/config/common.interface';
import { CountryAreaUnitEnum } from 'enums/country/area-unit.enum';
import { CountryCodeEnum } from 'enums/country/code.enum';
import { CountryCurrencyEnum } from 'enums/country/currency.enum';
import { LanguageCodeEnum } from 'enums/language/code.enum';

export const configCommon: ConfigCommonInterface = {
  countryCode: CountryCodeEnum.eg,
  areaUnit: CountryAreaUnitEnum.sqm,
  currencyCode: CountryCurrencyEnum.egp,
  language: {
    current: LanguageCodeEnum.en,
    alternative: LanguageCodeEnum.ar,
  },
};
