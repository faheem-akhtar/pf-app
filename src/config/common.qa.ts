import { CountryAreaUnitEnum } from 'enums/country/area-unit.enum';
import { CountryCurrencyEnum } from 'enums/country/currency.enum';
import { LanguageCodeEnum } from 'enums/language/code.enum';

import { ConfigCommonInterface } from './common.interface';

export const configCommon: ConfigCommonInterface = {
  areaUnit: CountryAreaUnitEnum.sqm,
  currencyCode: CountryCurrencyEnum.qar,
  countryCode: 'qa',
  insertDefaultLanguage: true,
  language: {
    current: LanguageCodeEnum.en,
    alternative: LanguageCodeEnum.ar,
  },
};
