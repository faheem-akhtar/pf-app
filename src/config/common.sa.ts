import { CountryAreaUnitEnum } from 'enums/country/area-unit.enum';
import { CountryCurrencyEnum } from 'enums/country/currency.enum';
import { LanguageCodeEnum } from 'enums/language/code.enum';
import { ConfigCommonInterface } from 'types/config/common.interface';

export const configCommon: ConfigCommonInterface = {
  areaUnit: CountryAreaUnitEnum.sqm,
  currencyCode: CountryCurrencyEnum.sar,
  countryCode: 'sa',
  language: {
    current: LanguageCodeEnum.ar,
    alternative: LanguageCodeEnum.en,
  },
};
