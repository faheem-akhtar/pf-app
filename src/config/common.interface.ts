import { CountryAreaUnitEnum } from 'enums/country/area-unit.enum';
import { CountryCurrencyEnum } from 'enums/country/currency.enum';
import { LanguageConfigInterface } from 'types/language/config.interface';

export interface ConfigCommonInterface {
  /**
   * area unit
   */
  areaUnit: CountryAreaUnitEnum;

  /**
   * currency code
   */
  currencyCode: CountryCurrencyEnum;

  /**
   * country code
   */
  countryCode: string;

  /**
   * language interface for country
   */
  language: LanguageConfigInterface;

  /**
   * Whether to insert default language in the url or not
   * @default true
   */
  insertDefaultLanguage: boolean;
}
