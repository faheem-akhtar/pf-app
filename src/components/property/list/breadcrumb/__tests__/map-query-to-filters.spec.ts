import { configCommon } from 'config/common';

import { breadcrumbMapQueryToFilters } from '../map-query-to-filters';

describe('breadcrumbMapQueryToFilters()', () => {
  it('should map correctly when locale is current', () => {
    expect(breadcrumbMapQueryToFilters(configCommon.language.current, '/en/search?l=6&t=1&c=1')).toMatchSnapshot();
  });

  it('should map correctly when locale is alternative', () => {
    expect(breadcrumbMapQueryToFilters(configCommon.language.alternative, '/ar/search?l=6&t=1&c=1')).toMatchSnapshot();
  });

  it('should map correctly for multiple locations when locale is current', () => {
    expect(breadcrumbMapQueryToFilters(configCommon.language.current, '/en/search?l=6-1231&t=1&c=1')).toMatchSnapshot();
  });

  it('should map correctly for multiple locations when locale is alternative', () => {
    expect(
      breadcrumbMapQueryToFilters(configCommon.language.alternative, '/ar/search?l=6-1231&t=1&c=1')
    ).toMatchSnapshot();
  });
});
