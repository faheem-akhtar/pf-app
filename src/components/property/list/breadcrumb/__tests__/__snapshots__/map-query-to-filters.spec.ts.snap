// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`breadcrumbMapQueryToFilters() should map correctly for multiple locations when locale is alternative 1`] = `
Object {
  "filter[category_id]": "1",
  "filter[locations_ids]": Array [
    Object {
      "abbreviation": "",
      "id": "6",
      "name": "أبوظبي",
      "path": "أبوظبي",
      "path_name": "",
    },
    Object {
      "abbreviation": "",
      "id": "1231",
      "name": "البرج",
      "path": "البرج",
      "path_name": "دبي, شارع الشيخ زايد",
    },
  ],
  "filter[property_type_id]": "1",
}
`;

exports[`breadcrumbMapQueryToFilters() should map correctly for multiple locations when locale is current 1`] = `
Object {
  "filter[category_id]": "1",
  "filter[locations_ids]": Array [
    Object {
      "abbreviation": "",
      "id": "6",
      "name": "Abu Dhabi",
      "path": "Abu Dhabi",
      "path_name": "",
    },
    Object {
      "abbreviation": "",
      "id": "1231",
      "name": "The Tower",
      "path": "The Tower",
      "path_name": "Dubai, Sheikh Zayed Road",
    },
  ],
  "filter[property_type_id]": "1",
}
`;

exports[`breadcrumbMapQueryToFilters() should map correctly when locale is alternative 1`] = `
Object {
  "filter[category_id]": "1",
  "filter[locations_ids]": Array [
    Object {
      "abbreviation": "",
      "id": "6",
      "name": "أبوظبي",
      "path": "أبوظبي",
      "path_name": "",
    },
  ],
  "filter[property_type_id]": "1",
}
`;

exports[`breadcrumbMapQueryToFilters() should map correctly when locale is current 1`] = `
Object {
  "filter[category_id]": "1",
  "filter[locations_ids]": Array [
    Object {
      "abbreviation": "",
      "id": "6",
      "name": "Abu Dhabi",
      "path": "Abu Dhabi",
      "path_name": "",
    },
  ],
  "filter[property_type_id]": "1",
}
`;
